frontend/lib/resolution/scope-queries.cpp

```diff
@@ -418,17 +438,39 @@ static const Scope* const& scopeForAutoModule(Context* context) {
   return QUERY_END(result);
 }
 
-static bool doLookupInImportsAndUses(Context* context,
-                                     const Scope* scope,
-                                     const ResolvedVisibilityScope* resolving,
-                                     const ResolvedVisibilityScope* cur,
-                                     UniqueString name,
-                                     bool onlyInnermost,
-                                     bool skipPrivateVisibilities,
-                                     VisibilitySymbols::ShadowScope shadowScope,
-                                     NamedScopeSet& checkedScopes,
-                                     std::vector<BorrowedIdsWithName>& result,
-                                     const VisibilitySymbols** foundVsOut) {
+static VisibilityStmtKind
+getKindForVisibilityClauseId(Context* context, ID visibilityClauseId) {
+  for (ID cur = parsing::idToParentId(context, visibilityClauseId);
+       !cur.isEmpty();
+       cur = parsing::idToParentId(context, cur)) {
+    auto tag = parsing::idToTag(context, cur);
+    if (asttags::isUse(tag)) {
+      return VIS_USE;
+      break;
+    } else if (asttags::isImport(tag)) {
+      return VIS_IMPORT;
+    }
+  }
+
+  CHPL_ASSERT(false && "should not be reached");
+  return VIS_USE;
+}
+
+static bool
+doLookupInImportsAndUses(Context* context,
+                         const Scope* scope,
+                         const ResolvedVisibilityScope* resolving,
+                         const ResolvedVisibilityScope* cur,
+                         UniqueString name,
+                         bool onlyInnermost,
+                         bool skipPrivateVisibilities,
+                         VisibilitySymbols::ShadowScope shadowScope,
+                         NamedScopeSet& checkedScopes,
+                         std::vector<BorrowedIdsWithName>& result,
+                         bool& foundExternBlock,
+                         std::vector<VisibilityTraceElt>* traceCurPath,
+                         std::vector<ResultVisibilityTrace>* traceResult) {
```

Comment:
How many of these arguments do not change between recursive invocations? The list of args is making me think that we should have a wrapper struct that contains all the unchanging args, to avoid feeding them through every single function.

Some of the things that I think don't change and could be put in the struct:
* Context
* `resolving`
* `onlyInnermost`
* `checkedScopes`
* `result`
* `foundExternBlock`
* `traceCurPath`
* `traceResult`

Reply:
I completely agree however I want to leave it as future work & the PR message already says this

> Future Work:
>
>  ...
> change doLookupIn... functions to work with a struct instead of too many arguments.